sound/soc/sof/pm.c

```diff
@@ -250,6 +250,21 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 	return ret;
 }
 
+int snd_sof_dsp_power_down_notify(struct snd_sof_dev *sdev)
+{
+	int ret;
+	/* Notify DSP of upcoming power down */
+	ret = sof_send_pm_ctx_ipc(sdev, SOF_IPC_PM_CTX_SAVE);
+	if (ret < 0) {
+		dev_err(sdev->dev,
+			"error %d: failed to send PM_CTX_SAVE message\n",
+			ret);
+	}
+
+	return ret;
+}
+EXPORT_SYMBOL(snd_sof_dsp_power_down_notify);
```

Comment:
This function is not called outside of core.c and pm.c. So no need to export it. In fact I'd just declare sof_send_pm_ctx_ipc() in sof-priv.h and call it with CTX_SAVE as the parameter in snd_sof_device_remove().

Reply:
I answered above, there is one reason we should keep this function and not export just sof_send_pm_ctx_ipc()  - there will likely be additional logic as to when we are sending this IPC.

Reply:
@mrajwa I agree with @ranj063 - if there are no users of this function currently outside of the module and no clear certain plans for such users in the near future and this function doesn't belong to a consistent exported API, I'd rather not export it for now. We can always export it later if a need arises.

Reply:
Let's by clear, we have to send IPC do the DSP at the very beginning of snd_sof_device_remove() - do you agree?
Now, you don't want to export this newly created function `snd_sof_dsp_power_down_notify()` so then what would you like to export? If the answer is `sof_send_pm_ctx_ipc()` and call it with the `SOF_IPC_PM_CTX_SAVE` argument then I am against it as to me it exposes low level / specialized functions. Second the `SOF_IPC_PM_CTX_SAVE` is not supported for all platforms so we have to wrap this IPC message with a condition like: 
`if (sof_ops(sdev)->remove) {
		ret = sof_send_pm_ctx_ipc(sdev, SOF_IPC_PM_CTX_SAVE);`

This way we will send IPC only for platform which support .remove procedure. Does it make sense to you?

Reply:
@mrajwa `snd_sof_device_remove()` is in core.c and `snd_sof_dsp_power_down_notify()` is in pm.c, both those files are linked together in the same module snd-sof.ko, so, you don't need `EXPORT_SYMBOL()` for it unless you also plan to call that function from other kernel modules.

Reply:
Yes no need to export I agree.
Updated.

I see all testes for check-kmod-load-unload.sh have passed.